src/csharp/Microsoft.Spark/ML/Feature/Bucketizer.cs

```diff
@@ -20,7 +20,7 @@ namespace Microsoft.Spark.ML.Feature
     /// will be thrown. The splits parameter is only used for single column usage, and splitsArray
     /// is for multiple columns.
     /// </summary>
-    public class Bucketizer : FeatureBase<Bucketizer>
+    public class Bucketizer : ScalaModel<Bucketizer>, ScalaMLWritable, ScalaMLReadable<Bucketizer>
```

Comment:
Could you explain the reason and change from using `FeatureBase`?

Reply:
Because FeatureBase is identifying all features as the same base class, but to be consistent with original scala version, the `real base` is `Params`. Based on Params we have Transformer, Model, Estimator and Evaluator, and that's why I added those base classes in `Base.cs` file. And by this change, we can clearly utilize the properties of those classes, like `Transformers` should implement `transform` and `transformSchema` methods, and `Estimators` should implement `fit` method, and `Models` inherits from `Transformers`; also we defined the type parameter restriction on those corresponding base classes. Besides, by adding `ScalaMLWritable` and `ScalaMLReadable` we're clearly spliting the behavior of base classes and ReadWrite functions. So only classes inherits these two classes should support read/write methods, which is much clearer than before.